### PR TITLE
Pataphysical Incursion (New Event and 2 Antags)

### PR DESCRIPTION
**Pull request by AndJusticeForAll0**
## About the Pull Request

Adds a new random event called "Pataphysical Incursion" that chooses from a pool of 3 (with more to come) antagonistics that rely on Pataphysics. These antags are Murphy Law (SCP-3143), a sterotypical film noir detective, and Don Quixote (SCP-4028),, a famous character in literature.  Shark Punching Centre Operative can also be chosen.
## Why It's Good For The Game

More random events are always good, and this one also adds a few new SCPs to the mix.
## Changelog

:cl:
add: Added Pataphysical Incursion event
add: Added Don Quixote and Murphy Law
add: Added Shark Punching Centre Operative
add: Office for Murphy Law on Z-Level 3
change: Events set to enable by default, power events completely disabled
/:cl:

### CODE REVIEW

**Comment by AndJusticeForAll0** [comment]
Potentially game breaking event chances were also changed

**Comment by AndJusticeForAll0** [comment]
Shark Punching Centre Operative has also been added as another possible event option.

**Comment by AndJusticeForAll0** [comment]
Events were also enabled because they weren't initially.

File: code/modules/events/event_container.dm [hunk 1]
```diff
@@ -150,17 +150,18 @@ var/global/list/severity_to_string = list(EVENT_LEVEL_MUNDANE = "Mundane", EVENT
...
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Appendicitis", 						/datum/event/spontaneous_appendicitis, 	0,		list(ASSIGNMENT_MEDICAL = 10), 1),
-		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Communication Blackout",				/datum/event/communications_blackout,	50,		list(ASSIGNMENT_AI = 100, ASSIGNMENT_ENGINEER = 20)),
+		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Communication Blackout",				/datum/event/communications_blackout,	35,		list(ASSIGNMENT_AI = 100, ASSIGNMENT_ENGINEER = 20)),
 		new /datum/event_meta/no_overmap(EVENT_LEVEL_MODERATE, "Electrical Storm",			/datum/event/electrical_storm, 			10,		list(ASSIGNMENT_ENGINEER = 15, ASSIGNMENT_JANITOR = 10)),
 //		new /datum/event_meta/no_overmap(EVENT_LEVEL_MODERATE, "Gravity Failure",			/datum/event/gravity,	 				50,		list(ASSIGNMENT_ENGINEER = 20)),
-		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Grid Check",							/datum/event/grid_check, 				40,		list(ASSIGNMENT_ENGINEER = 15, ASSIGNMENT_ANY = 5)),
+		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Grid Check",							/datum/event/grid_check, 				0,		list(ASSIGNMENT_ENGINEER = 15, ASSIGNMENT_ANY = 5)),
```
Comment (MACIEKBAKI) [inline code/modules/events/event_container.dm:158]: you ought to note changes like you made in this file in the PR desc

**Review by MACIEKBAKI — DISMISSED** [review]

**Comment by MACIEKBAKI** [comment]
please also add to the pr desc what map changes you've made, i'm too lazy to review that

**Comment by MACIEKBAKI** [comment]
also fix up this space tile
image

**Comment by MACIEKBAKI** [comment]
overall, pretty nice PR, save for those few small details

File: code/game/antagonist/outsider/donquixote.dm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	player.current.real_name = player.name
+	player.current.SetName(player.current.name)
+
+	var/mob/living/carbon/human/H = player.current
+	if(istype(H))
+		H.gender = MALE
+		H.age = rand(25,45)
+		H.dna.ready_dna(H)
+
+	return
```
Comment (ivanmixo) [inline code/game/antagonist/outsider/donquixote.dm]: Pointless return. Remove it on this and all the other ones. I know the others do it like this but they're shitty and should be addressed too.

File: code/game/objects/items/weapons/cards_ids.dm [hunk 3]
```diff
@@ -401,6 +401,18 @@ var/const/NO_EMAG_ACT = -50
...
 
+/obj/item/card/id/murphylaw
+	name = "private investigator's ID"
+	desc = "The ID of a hardboiled private investigator."
+	item_state = "gold_id"
+	registered_name = "Investigator"
+	assignment = "Investigator"
+	detail_color = COLOR_AMBER
+
+/obj/item/card/id/murphylaw/New()
```
Comment (ivanmixo) [inline code/game/objects/items/weapons/cards_ids.dm]: Use `Initialize()`, not `New()`. Also `return ..()` 2 lines down.

File: code/modules/events/pataphysics.dm [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+	command_announcement.Announce("There has been a recent spike in narrative energy within the area of Site-53, prepare for a pataphysical narrative layer incursion.")
+
+/datum/event/pataphysics/start()
+	var/list/valid_types = list()
+
+	for(var/antag_type in GLOB.all_antag_types_)
+		var/datum/antagonist/antag = GLOB.all_antag_types_[antag_type]
+		if(antag.flags & ANTAG_PATAPHYSICS)
+			valid_types |= antag
+	if(valid_types.len)
```
Comment (ivanmixo) [inline code/modules/events/pataphysics.dm]:
```suggestion
	if(length(valid_types))
```

File: code/game/antagonist/outsider/donquixote.dm [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+/datum/antagonist/donquixote/equip(var/mob/living/carbon/human/player)
+	player.add_language(LANGUAGE_ENGLISH)
+	player.equip_to_slot_or_del(new /obj/item/clothing/under/bluetunic(player), slot_w_uniform)
+	player.equip_to_slot_or_del(new /obj/item/clothing/shoes/jackboots/medievalboots(player), slot_shoes)
+	player.equip_to_slot_or_del(new /obj/item/clothing/suit/champarmor(player), slot_wear_suit)
+	player.equip_to_slot_or_del(new /obj/item/material/sword(player), slot_belt)
+
+
+/datum/antagonist/donquixote/update_antag_mob(var/datum/mind/player)
+
```
Comment (ivanmixo) [inline code/game/antagonist/outsider/donquixote.dm]: Remove this space too, guh.

**Review by ivanmixo — CHANGES_REQUESTED** [review]

**Comment by AndJusticeForAll0** [comment]
All issues have been fixed.

File: code/game/antagonist/outsider/murphylaw.dm [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	player.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/det_trench(player), slot_wear_suit)
+	player.equip_to_slot_or_del(new /obj/item/clothing/glasses/sunglasses(player), slot_glasses)
+	player.equip_to_slot_or_del(new /obj/item/storage/briefcase/crimekit, slot_in_backpack)
+	player.equip_to_slot_or_del(new /obj/item/storage/backpack, slot_back)
+	player.equip_to_slot_or_del(new /obj/item/storage/belt/holster/security/fullmateba(player), slot_belt)
+
+	create_id("Private Investigator", player)
+
+/datum/antagonist/murphylaw/update_antag_mob(var/datum/mind/player)
+
```
Comment (ivanmixo) [inline code/game/antagonist/outsider/murphylaw.dm]: These too.
```suggestion
```

File: code/game/antagonist/outsider/sharkpunchingcentre.dm [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	player.equip_to_slot_or_del(new /obj/item/device/radio/headset/ert(player), slot_l_ear)
+	player.equip_to_slot_or_del(new /obj/item/clothing/under/ert(player), slot_w_uniform)
+	player.equip_to_slot_or_del(new /obj/item/clothing/shoes/jackboots(player), slot_shoes)
+	player.equip_to_slot_or_del(new /obj/item/clothing/gloves/boxing(player), slot_gloves)
+	player.equip_to_slot_or_del(new /obj/item/clothing/head/helmet/scp/eta(player), slot_head)
+	player.equip_to_slot_or_del(new /obj/item/clothing/suit/armor/vest/scp/medarmor/eta(player), slot_wear_suit)
+
+
+/datum/antagonist/sharkpunchingcentre/update_antag_mob(var/datum/mind/player)
+
```
Comment (ivanmixo) [inline code/game/antagonist/outsider/sharkpunchingcentre.dm]:
```suggestion
```

**Review by ivanmixo — DISMISSED** [review]
Code side okay, head map guy needs to see map changes.

**Review by hry-gh — APPROVED** [review]
mapping

**Review by ivanmixo — DISMISSED** [review]
The reviews were *not* stale, actually.

**Review by Tsurupeta — CHANGES_REQUESTED** [review]
It doesn't even work how it's supposed to

**Comment by aidan-bowman** [comment]
reminder that nobody can make any changes to the event system while this is open (at least without major merge conflict headaches)

**Comment by Tsurupeta** [comment]
It's not like it's getting fixed anytime soon. Closing due to being stale.

**Pull request closed without merging**